Source/TestDafny/MultiBackendTest.cs

```diff
@@ -94,33 +94,32 @@ public class MultiBackendTest {
     output.WriteLine("Verifying...");
 
     var (exitCode, outputString, error) = RunDafny(options.DafnyCliPath, dafnyArgs);
-    if (exitCode != 0) {
-      output.WriteLine("Verification failed. Output:");
-      output.WriteLine(outputString);
-      output.WriteLine("Error:");
-      output.WriteLine(error);
-      return exitCode;
-    }
+    // If there is a .verifier.expect file, then we expect the output to match the .verifier.expect file contents.
+    // In either case, we expect verification to return a 0 exit code.
     var expectFileForVerifier = $"{options.TestFile}.verifier.expect";
     if (File.Exists(expectFileForVerifier)) {
       var expectedOutput = File.ReadAllText(expectFileForVerifier);
       // Chop off the "Dafny program verifier finished with..." trailer
       var trailer = new Regex("\r?\nDafny program verifier[^\r\n]*\r?\n").Match(outputString);
       var actualOutput = outputString.Remove(trailer.Index, trailer.Length);
       var diffMessage = AssertWithDiff.GetDiffMessage(expectedOutput, actualOutput);
-      if (diffMessage == null) {
-        return 0;
+      if (diffMessage != null) {
+        output.WriteLine(diffMessage);
+        return 1;
       }
-
-      output.WriteLine(diffMessage);
-      return 1;
+    }
+    if (exitCode != 0) {
```

Comment:
Just to be clear, this change only means that if the output and exit code don't match, we'll report the former instead of the latter, correct? That's just fine with me if it happens to be useful for you.

Reply:
I forgot to implement and test that the output is empty if there is no `.verifier.expect` file. Now done.